packages/destination-actions/src/destinations/mixpanel/incrementProperties/index.ts

```
@@ -0,0 +1,89 @@
+      default: {
+        '@path': '$.context.ip'
+      }
+    },
+    user_id: {
+      label: 'User ID',
+      type: 'string',
+      allowNull: true,
+      description: 'The unique user identifier set by you',
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    anonymous_id: {
+      label: 'Anonymous ID',
+      type: 'string',
+      allowNull: true,
+      description: 'The generated anonymous ID for the user',
+      default: {
+        '@path': '$.anonymousId'
+      }
+    },
+    increment: {
+      label: 'Increment Numerical Properties',
+      type: 'object',
+      description: 'Object of properties and the values to increment or decrement',
+      multiple: false,
+      defaultObjectUI: 'keyvalue',
+      default: {
+        '@path': '$.properties.increment'
+      }
+    }
+  },
+
+  perform: async (request, { payload, settings }) => {
+    if (!settings.projectToken) {
+      throw new IntegrationError('Missing project token', 'Missing required field', 400)
+    }
+
+    const apiServerUrl = getApiServerUrl(settings.apiRegion)
+
+    const responses = []
+
+    if (payload.increment && Object.keys(payload.increment).length > 0) {
+      const data: MixpanelEngageProperties = {
+        $token: settings.projectToken,
+        $distinct_id: payload.user_id ?? payload.anonymous_id,
+        $ip: payload.ip
+      }
+      data.$add = {}
+
+      for (const key of Object.keys(payload.increment)) {
+        const value = payload.increment[key]
+        if (typeof value === 'string' || typeof value === 'number') {
+          if (!isNaN(+value)) {
+            data.$add[key] = +value
+          }
+        }
+      }
+
+      const response = request(`${apiServerUrl}/engage`, {
+        method: 'post',
+        body: new URLSearchParams({ data: JSON.stringify(data) })
+      })
+
+      responses.push(response)
+    }
+
+    return Promise.all(responses)
```

I wonder should we put a hard limit on the number of requests which can be made? Say 10 or 15?
If you do decide on a limit, let's throw an error if the customer provides more than number of key:value pairs.